admittance_controller/src/incremental_ik_calculator.cpp

```diff
@@ -74,8 +73,17 @@ bool IncrementalIKCalculator::convertCartesianDeltasToJointDeltas(Eigen::VectorX
   svd_ = Eigen::JacobiSVD<Eigen::MatrixXd>(jacobian_, Eigen::ComputeThinU | Eigen::ComputeThinV);
   matrix_s_ = svd_.singularValues().asDiagonal();
   pseudo_inverse_ = svd_.matrixV() * matrix_s_.inverse() * svd_.matrixU().transpose();
```

Comment:
My two cents on this part of the code. If you are going through a singularity, basically if the "geometrical/configuration Jacobian" is near losing a rank,  the singular values will be going near zero, and the inverse will generate extremely high speed. The usual way is to fix this problem is to set to zero the speed of the joint which is going through the singularity. If you have a powerful robot, clamping the speed will not be enough.

Reply:
What about clamping acceleration? We have the capacity to do that fairly easily with Reflexxes.

Reply:
A simple implementation will be to iterate over the diagonal element of matrix_s_ and compute the following:
`pinv_matrix_s_(i,i) = matrix_s_(i,i) /( matrix_s_(i,i) * matrix_s_(i,i) + rho )`
with rho a regularisation parameter to be defined by the user.

You can find a tutorial on damped least square here:
https://link.springer.com/content/pdf/10.1007/BF01254007.pdf

Two notes on this: 
1/ This implies that you will not be following exactly the admittance behavior near a singularity. 
2/ many consider that formulating the pseudo inverse as an optimization problem is making its incremental extension easier.

For instance if you want to take into account the robot limits it is much more easier when written as an optimization problem.

@AndyZe this is a good idea. As far as I understood the Reflexx library it allows to define the trajectory dynamics in the task space, but I am not sure that the robot dynamics is taken into account. If not this is fine as long as you do not go very fast. The faster you will go, the more important is to take into account the robot dynamics. Clamping the acceleration may not be always the most appropriate, it can be too late and too constraining depending if you are near a limit or not. Reflexx was started/written by Kroger Tarsten who was the head of IAR-PR. @destogl may have more precise comments than me on this point as he recently graduated from this team. (Congrats by the way @destogl , even if this is quite late).

Reply:
Fyi I am working on pulling in the [servo approach](https://github.com/ros-planning/moveit2/blob/7064da6e4b714481047fe55803e4bca0bbf32ada/moveit_ros/moveit_servo/src/servo_calcs.cpp#L696) for this into the admittance controller.